python/pyarrow/tests/test_array.py

```diff
@@ -3579,6 +3579,42 @@ def test_run_end_encoded_from_buffers():
                                            1, offset, children)
 
 
+def test_run_end_encoded_from_array_with_type():
+    run_ends = pa.array([1, 3, 6], type=pa.int32())
+    values = pa.array([1, 2, 3], type=pa.int64())
+
+    ree_type = pa.run_end_encoded(pa.int32(), pa.int64())
+
+    arr = [1, 2, 2, 3, 3, 3]
+    result = pa.array(arr, type=ree_type)
+
+    assert result.run_ends.equals(run_ends)
+    assert result.values.equals(values)
```

Comment:
Instead of each time checking both run ends and values, I think you could also once created the expected result (`expected = pa.RunEndEncodedArray.from_arrays(run_ends, values, ree_type)`), and then just check `assert result.equals(expected)`